shiftcommerce/shift-next
# Trendy Issue: Enable Brotli Compression [WIP]

## Overview

Github Issue: https://github.com/shiftcommerce/trendy-golf/issues/60

This PR enables brotli and gzip compression via the `shrink-ray-current` library. Ideally, this should be replaced with `compression js` when this issue is resolved - https://github.com/expressjs/compression/issues/71

PR also bumps the package version to `v0.14.0`. 

#### Caveat: 
The `shrink-ray-current` library utilises `node-zopfli-es` when gzipping static file. However, this is an option dependency and since it is ([~100x slower than gzip](https://www.npmjs.com/package/node-zopfli-es)) we have not included it. As such, we will see some warnings in the console eg. -

```
$ NODE_ENV=production node server/server.js
(node:6323) [ZOPFLI_COMPAT] MISSING_MODULE: Module "node-zopfli-es" was unavailable
Zopfli compression unavailable; will fall back to gzip.
> Ready on http://localhost:3000
```

## QA

1. Checkout Shift Frontend React branch `trendy-issue/60_enable_brotli2` and boot app

2. Navigate to `http://localhost:3000`

3. In the chrome developer tools - network tab, check that the `Content-Encoding` is `br`

![Brotli compression](https://user-images.githubusercontent.com/4486874/58257792-73d6b600-7d69-11e9-91ff-26364549a5cc.png)
